crates/starknet-server/src/main.rs

```diff
@@ -77,7 +99,19 @@ async fn main() -> Result<(), anyhow::Error> {
     info!("Starknet Devnet listening on {}", addr);
 
     // spawn the server on a new task
-    let serve = tokio::task::spawn(server);
+    let serve = tokio::task::spawn(server.with_graceful_shutdown(shutdown_signal(api.clone())));
 
     Ok(serve.await??)
 }
+
+pub async fn shutdown_signal(api: Api) {
+    tokio::signal::ctrl_c().await.expect("Failed to install CTRL+C signal handler");
+
+    // Wait for the CTRL+C signal
+    signal::ctrl_c().await.expect("Failed to read CTRL+C signal");
+
+    let starknet = api.starknet.read().await;
+    if starknet.config.dump_on.is_some() && starknet.config.dump_on == Some(DumpMode::OnExit) {
```

Comment:
```suggestion
    if starknet.config.dump_on == Some(DumpMode::OnExit) {
```

Reply:
Fixed in 2e2312e9305304bcc68c0c2858ff094e6edf5a91